File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>m;
        for(auto key : nums)
        {
            m[key]++;
        }
        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>>pq;
        for(auto &key : m)
        {
            pq.push({key.second,key.first});
            if(pq.size()>k)
            {
                pq.pop();
            }
        }
        vector<int>res;
        while(pq.empty()==false)
        {
            auto curr = pq.top();
            pq.pop();
            res.push_back(curr.second);
        }
        return res;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        
        HashMap<Integer , Integer> m = new HashMap<>();
        
        for(int i=0;i<nums.length;i++){
            int freq = m.getOrDefault(nums[i],0);
            m.put(nums[i],freq+1);
        }
        
        PriorityQueue<Pair<Integer, Integer>> pq = new PriorityQueue<>((a,b)-> a.getKey() - b.getKey());
        
        
        for(Integer key : m.keySet()){
            int freq = m.get(key);
            pq.offer(new Pair<Integer,Integer>(freq,key));
            
            if(pq.size() > k){
                pq.poll();
            }
            
        }
        
        int []res = new int[k];
        int j=0;
        while(pq.size() != 0){
            Pair<Integer,Integer> top = pq.peek();
            res[j] = top.getValue();
            j++;
            pq.poll();
        }
        
        return res;
        
    }
}